dadjokes-onchain/foundry/src/DadJokes.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract DadJokes {
+    struct Joke {
+        string setup;
+        string punchline;
+        address creator;
+        uint256 classicRewards;
+        uint256 funnyRewards;
+        uint256 groanerRewards;
+        bool isDeleted;
+    }
+
+    Joke[] private jokes;
+    mapping(address => uint256) public creatorBalances;
+
+    uint256 private constant CLASSIC_REWARD = 0.001 ether;
+    uint256 private constant FUNNY_REWARD = 0.005 ether;
+    uint256 private constant GROANER_REWARD = 0.01 ether;
+
+    event JokeAdded(uint256 indexed jokeId, address indexed creator);
+    event JokeRewarded(
+        uint256 indexed jokeId,
+        uint8 rewardType,
+        uint256 rewardAmount
+    );
+    event JokeDeleted(uint256 indexed jokeId);
+    event BalanceWithdrawn(address indexed creator, uint256 amount);
+
+    function addJoke(string memory _setup, string memory _punchline) public {
+        uint256 jokeId = jokes.length;
+        jokes.push(Joke(_setup, _punchline, msg.sender, 0, 0, 0, false));
+        emit JokeAdded(jokeId, msg.sender);
+    }
+
+    function getJokes() public view returns (Joke[] memory) {
+        return jokes;
+    }
+
+    function rewardJoke(uint256 _index, uint8 _rewardType) public payable {
+        require(_index < jokes.length, "Invalid joke index");
```

Comment:
does this need its own require? would the default out of bound error not suffice?

Reply:
The default error would work. This error is more user-friendly.